_source/_posts/2022-07-11-kubernetes-jhipster-aws.md

```diff
@@ -0,0 +1,730 @@
+---
+layout: blog_post
+title: "JHipster Microservices on AWS with Amazon Elastic Kubernetes Service"
+author: andrew-hughes
+by: contractor
+communities: [java,devops]
+description: "Build a JHipster Microservice on AWS with Amazon Elastic Kubernetes Service and secure it with OAuth 2.0 and OIDC using Okta as a provider"
+tags: [java,jhipster,kubernetes,k8s,devops,microservices]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+
+---
+
+In this tutorial, you'll use JHipster to build a microservice stack and deploy it to Amazon Elastic Kubernetes Service (EKS). The microservices will use Java and Spring Boot for resource servers and Vue for the frontend. The server-side applications will use Spring's Reactive technology stack. The microservice will include multiple databases and database types: PostgresSQL, MongoDB, and Neo4j. You'll secure the service using OAuth 2.0 and OpenID Connect (OIDC) using Okta as the OIDC provider. With Kubernetes secrets and `kubeseal`, you'll encrypt all of the secrets in the project configuration files. To generate the project, you'll use JHipster CLI and the JHipster Domain Language (JDL).
+
+**Prerequisites**
+
+Before you get started, install the required software listed below. You'll need to sign up for an Amazon Web Services account and a free Okta account (which you will be able to sign up for later using the Okta CLI). The Kubernetes cluster required to finish this tutorial **does not qualify for free tier on AWS**. However, **if you are careful about stopping and deleting the cluster** when not actively working on the tutorial, the cost should be very small, only a few dollars.
+
+- [Docker](https://docs.docker.com/get-docker/): you’ll need to have both **Docker Engine** and **Docker Compose** installed (If you install the Docker desktop, this will automatically install both. On Linux, if you install Docker Engine individually, you will have to also [install Docker Compose](https://docs.docker.com/compose/install/)) separately.
```

Comment:
```suggestion
- [Docker](https://docs.docker.com/get-docker/): you’ll need to have both **Docker Engine** and **Docker Compose** installed. If you install the Docker desktop, this will automatically install both. On Linux, if you install Docker Engine individually, you will have to also [install Docker Compose](https://docs.docker.com/compose/install/) separately.
```
removed the extra parens.

Reply:
That's actually a mistake. I need to fix it. They don't need docker compose for this tutorial.